[[ Bug 19543 ]] Dictionary improperly parses hard wrapped lines

bwmilby opened the pull request:
When a paragraph ends with a colon and the last word falls on a new line then that word is parsed as a new element.  Due to the way the regex is parsed, something inside parenthesis before the colon could also trigger this bug.

The proposed fix:  Unless there is a preceding blank line, assume that this term is part of the existing content.  If there is no content, then assume it is indeed a new element.  An exception is if the element contains the content on the same line, in that case a new element can start on the next line.  Some LCB documentation has "Description:" immediately following another multi-line element without a blank line (i.e. examples).  Added a check that can be extended if other elements are found that need to be treated as an exception.

I ran a diff on the built_api.js from before my change and after.  There are 584 lines of changes.  Not all of them are evident in the dictionary viewer.  (Only the first 128 lines are for ~32 actual changes, the rest were due to reordering of svgpath and svgspinner in the JSON.)

Some entries fixed by the change:
- open printing
- open printing to pdf
- switch

bwmilby committed:
[[ Bug 19543 ]] Dictionary improperly parses hard wrapped lines where word is followed by colon

When a paragraph ends with a colon and the last word falls on a new line then that word is parsed as a new element.  Due to the way the regex is parsed, something inside parenthesis before the colon could also trigger this bug.

The proposed fix:  Unless there is a preceding blank line, assume that this term is part of the existing content.  If there is no content, then assume it is indeed a new element.  An exception is if the element contains the content on the same line, in that case a new element can start on the next line.  Some LCB documentation has "Description:" immediately following another multi-line element without a blank line (i.e. examples).  Added a check that can be extended if other elements are found that need to be treated as an exception.

I ran a diff on the built_api.js from before my change and after.  There are 584 lines of changes.  Not all of them are evident in the dictionary viewer.  (Only the first 128 lines are for ~32 actual changes, the rest were due to reordering of svgpath and svgspinner in the JSON.)

Some entries fixed by the change:
- open printing
- open printing to pdf
- switch

bwmilby committed:
[[Bug 19543]] Add bug fix note

livecodeali commented:
This looks good. I have one suggestion, which is to replace
```
--Always recognize certain elements: Description,
```
and 
```
",Description"
```
with a suitably named constant, eg `kAlwaysRecognizedElements` or something.

Also, is the empty string deliberately included in the items? I believe it can't be empty anyway because of the regex, but I just want to make sure.

bwmilby commented:
Empty string would not get used due to the regex so it isn’t needed (was copying a form I saw elsewhere).
I’ll update to use a constant.

bwmilby committed:
[[ Bug 19543 ]] Dictionary improperly parses hard wrapped lines

Use a constant for the list of elements that are always recognized as such.  Currently just one item.

livecodeali commented:
@livecode-vulcan review ok 349a29a

livecode-vulcan commented:
💙 review by @livecodeali ok 349a29aa31895188a6ca373ac4be197a97785925

livecode-vulcan commented:
😎 test success 349a29aa31895188a6ca373ac4be197a97785925

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1033)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1049)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1023)
* try-community-universal-ios-iphoneos11.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.1/builds/96)
* try-community-universal-ios-iphonesimulator11.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.1/builds/96)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1028)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1050)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1033)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1049)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1032)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1042)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1065)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1028)
